services/src/contexts/mod.rs

```diff
@@ -450,36 +447,21 @@ mod tests {
     #[serial]
     async fn write_model_test() {
         let cfg = get_config_element::<crate::util::config::MachineLearning>().unwrap();
-        let cfg_backup = cfg.model_defs_path;
+        std::fs::create_dir_all(&cfg.model_defs_path).unwrap();
 
-        let tmp_dir = tempfile::tempdir().unwrap();
-        let tmp_path = tmp_dir.path();
-        std::fs::create_dir_all(tmp_path.join("pro/ml/xgboost")).unwrap();
-
-        let temp_ml_path = tmp_path.join("pro/ml").to_str().unwrap().to_string();
-
-        set_config("machinelearning.model_defs_path", temp_ml_path).unwrap();
         let ctx = InMemoryContext::test_default()
             .default_session_context()
             .await;
         let mut exe_ctx = ctx.execution_context().unwrap();
 
-        let model_path = PathBuf::from("xgboost/model.json");
-
         exe_ctx
-            .write_ml_model(model_path, String::from("model content"))
+            .write_ml_model(cfg.model_defs_path.clone(), String::from("model content"))
```

Comment:
same here, it should be a ModelId

Reply:
Implemented in 1168be2810c99a6bbee7b5b4d26596dee1373a5a

---

Commit msg:
refactor: implement a ModelId that is used in conjunction with ml models